graphql_compiler/tests/schema_transformation_tests/test_merge_schemas.py

```diff
@@ -456,23 +479,755 @@ def test_clashing_directives(self):
+                        field_name = 'identifier',
+                    ),
+                    out_edge_only=False,
+                ),
+            ]
+        )
+        merged_schema_string = dedent('''\
+            schema {
+              query: RootSchemaQuery
+            }
+
+            type RootSchemaQuery {
+              Human: Human
+              Person: Person
+            }
+
+            type Human {
+              id: String
+              out_example_edge: [Person] @stitch(source_field: "id", sink_field: "identifier")
+            }
+
+            type Person {
+              age: Int
+              name: String!
+              identifier: String
+              friends: [Person]
+              in_example_edge: [Human] @stitch(source_field: "identifier", sink_field: "id")
+            }
+        ''')
+        self.assertEqual(merged_schema_string, print_ast(merged_schema.schema_ast))
+
+    def test_invalid_edge_within_single_schema(self):
+        with self.assertRaises(InvalidCrossSchemaEdgeError):
+            merge_schemas(
+                OrderedDict([
+                    ('schema', parse(ISS.union_schema)),
+                ]),
+                [
+                    CrossSchemaEdgeDescriptor(
+                        edge_name = 'example_edge',
+                        outbound_side = FieldReference(
+                            schema_id = 'schema',
+                            type_name = 'Human',
+                            field_name = 'id',
+                        ),
+                        inbound_side = FieldReference(
+                            schema_id = 'schema',
+                            type_name = 'Droid',
+                            field_name = 'id',
+                        ),
+                        out_edge_only=False,
+                    ),
+                ]
+            )
+
+    def test_invalid_edge_nonexistent_schema(self):
+        with self.assertRaises(InvalidCrossSchemaEdgeError):
+            merge_schemas(
+                OrderedDict([
+                    ('first', parse(ISS.basic_schema)),
+                    ('second', parse(ISS.same_field_schema)),
+                ]),
+                [
+                    CrossSchemaEdgeDescriptor(
+                        edge_name = 'example_edge',
+                        outbound_side = FieldReference(
+                            schema_id = 'first',
+                            type_name = 'Human',
+                            field_name = 'id',
+                        ),
+                        inbound_side = FieldReference(
+                            schema_id = 'third',
+                            type_name = 'Person',
+                            field_name = 'identifier',
+                        ),
+                        out_edge_only=False,
+                    ),
+                ]
+            )
+
+    def test_invalid_edge_type_in_wrong_schema(self):
+        with self.assertRaises(InvalidCrossSchemaEdgeError):
```

Comment:
Doesn't this test overlap with `test_invalid_edge_nonexistent_type`? Seems like in both cases, we're looking for a nonexistent type in a schema.

Reply:
The actual execution of these two invalid cases ends up being the same (same flow, raises same error) through the code, but it could have potentially been the case (or perhaps will be the case in the future if the implementation changes) that types are stored and checked for differently, and not finding a schema has a different execution flow then finding a schema and not finding the specified type.